tests/mocha/insertion_marker_manager_test.js

```diff
@@ -0,0 +1,329 @@
+            const dxy = new Blockly.utils.Coordinate(0, 0);
+            const stub = stubComponentManager(this.workspace);
+            stub.withArgs('fakeDragTarget',
+                Blockly.ComponentManager.Capability.DELETE_AREA).returns(true);
+            const fakeDragTarget = {
+                wouldDelete: sinon.fake.returns(false),
+                id: 'fakeDragTarget',
+            };
+            this.manager.update(dxy, fakeDragTarget);
+            chai.assert.isFalse(this.manager.wouldDeleteBlock());
+            chai.assert.isTrue(fakeDragTarget.wouldDelete.called);
+        });
+
+        test('Drag target is not a delete area', function() {
+            const dxy = new Blockly.utils.Coordinate(0, 0);
+            const stub = stubComponentManager(this.workspace);
+            stub.withArgs('fakeDragTarget',
+                Blockly.ComponentManager.Capability.DELETE_AREA).returns(false);
+            const fakeDragTarget = {
+                wouldDelete: sinon.fake.returns(false),
+                id: 'fakeDragTarget',
+            };
+            this.manager.update(dxy, fakeDragTarget);
+            chai.assert.isFalse(this.manager.wouldDeleteBlock());
+            chai.assert.isFalse(fakeDragTarget.wouldDelete.called);
+        });
+
+        test('Not over drag target', function() {
+            const dxy = new Blockly.utils.Coordinate(0, 0);
+            this.manager.update(dxy, null);
+            chai.assert.isFalse(this.manager.wouldDeleteBlock());
+        });
+    });
+
+    suite('Would connect stack blocks', function() {
+        setup(function() {
+            const state = {
+                'blocks': {
+                  'blocks': [
+                    {
+                      'type': 'stack_block',
+                      'id': 'first',
+                      'x': 0,
+                      'y': 0,
+                    },
+                    {
+                      'type': 'stack_block',
+                      'id': 'other',
+                      'x': 200,
+                      'y': 200,
+                    },
+                  ],
+                },
+              };
+            Blockly.serialization.workspaces.load(state, this.workspace);
+            this.block = this.workspace.getBlockById('first');
+            this.block.setDragging(true);
+            this.manager = new Blockly.InsertionMarkerManager(this.block);
+        });
+
+        test('No other blocks nearby', function() {
+            this.manager.update(new Blockly.utils.Coordinate(0, 0), null);
+            chai.assert.isFalse(this.manager.wouldConnectBlock());
+        });
+
+        test('Near other block: above', function() {
+            this.manager.update(new Blockly.utils.Coordinate(200, 190), null);
+            chai.assert.isTrue(this.manager.wouldConnectBlock());
+            const markers = this.manager.getInsertionMarkers();
+            chai.assert.equal(markers.length, 1);
```

Comment:
What are these assertions checking? These look like dupes of the 'Create markers' assertions, can they  be removed?

Reply:
Checking that it's safe to call `markers[0].nextConnection.isConnected`. But I can remove the assertions on `markers.length` and assume that if I'm wrong the test will still fail on the next access. What do you think?

Reply:
Sounds good to me!